stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/startup/StartupFragment.kt

```diff
@@ -19,4 +30,44 @@ class StartupFragment : Fragment() {
             }
         )
     }
+
+    private fun showDeprecationWarningIfNeeded() {
+        val context = requireContext()
+
+        if (warningShown(context)) {
+            return
+        }
+
+        val checkbox = CheckBox(context).apply { text = "Do not show this again" }
+        val frame = FrameLayout(context).apply {
+            setPadding(Utils.dpToPx(16), Utils.dpToPx(12), Utils.dpToPx(16), 0)
+        }
+        frame.addView(checkbox)
+
+        AlertDialog.Builder(context)
+            .setTitle("Deprecation warning")
+            .setMessage(Html.fromHtml("This sample uses our old UI kit.<br/><br/>Take a look at the new UI Components implementation in the <b>ui-components</b> and <b>ui-components-sample</b> modules instead!"))
+            .setView(frame)
+            .setPositiveButton("Ok") { _, _ ->
+                if (checkbox.isChecked) {
+                    setWarningShown(context)
+                }
+            }
+            .show()
+    }
+
+    private fun warningShown(context: Context): Boolean {
+        return PreferenceManager.getDefaultSharedPreferences(context)
+            .getBoolean(OLD_SAMPLE_WARNING_SHOWN_KEY, false)
+    }
+
+    private fun setWarningShown(context: Context) {
+        PreferenceManager.getDefaultSharedPreferences(context).edit {
+            putBoolean(OLD_SAMPLE_WARNING_SHOWN_KEY, true)
+        }
```

Comment:
Does this work without `apply()`?

Reply:
Yes! https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:core/core-ktx/src/main/java/androidx/core/content/SharedPreferences.kt;l=39